### ISSUE TITLE
Warning: findDOMNode is deprecated and will be removed in the next major release.

### ISSUE THREAD

**Issue by RM-Eric** [react-grid-layout/react-grid-layout#2073]
### Describe the bug

I am using v1.4.4 of the library, and recently updated to React 18.3.1 from React 17. 
Now, when moving an item in the grid, I get this error in the console:
```
findDOMNode is deprecated and will be removed in the next major release. Instead, add a ref directly to the element you want to reference.
```

I have found other issues referencing this, but they all seem to indicate that it was fixed in v1.2.3

This is when using `Responsive` and `WidthProvider` such as:
```
const ResponsiveGridLayout = WidthProvider(Responsive);
```

I have tried passing a ref to the `ResponsiveGridLayout ` component, but I still get the error.

### Your Example Website or App

https://codesandbox.io/p/sandbox/smoosh-sea-sx3p43?file=%2Fsrc%2Findex.js

### Steps to Reproduce the Bug or Issue

Using React 18.3.1:

```
import { Responsive, WidthProvider } from "react-grid-layout";
const ResponsiveGridLayout = WidthProvider(Responsive);

//...

//Render the component:

<ResponsiveGridLayout 
   // ... props
/>
```

### Expected behavior

The error should not be thrown in console

### react-grid-layout library version

1.4.4

### Operating System Version

Windows

### Browser

Chrome

**Comment by salmanorak** [comment]
The error in the console is not about adding or not adding a ref to your ResponsiveGridLayout element. In the findDOMNode, it is a function to grab an element from the DOM tree. 

https://react.dev/reference/react-dom/findDOMNode

I think the behind-the-hood library uses the findDOMNode and react is giving a warning and it says this function will be removed with the next major release. So, it is an error in the library's internal logic. 

In conclusion this should be changed by library.

**Comment by salmanorak** [comment]
Also from the change logs I see that maintainers have mentioned how to avoid this error. You can reference to that example.

https://github.com/react-grid-layout/react-draggable/blob/a61bd5feaee52f753375ab1955cafe3881b2bc14/CHANGELOG.md?plain=1#L44

**Comment by STRML** [comment]
This is fixed in v2! 🎉

v2 is a complete rewrite using functional components and refs - no more `findDOMNode` usage.

```bash
npm install react-grid-layout@latest
```

The warning will be gone with v2.2.0+.
